modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -250,6 +340,41 @@ private void maybeExecuteRetention(ClusterState state, DataStream dataStream) {
                 );
             }
         }
+        return indicesToBeRemoved;
+    }
+
+    /*
+     * This method force merges any indices in the datastream that are not in the indicesToExclude set and that have not be force merged
+     * previously. Before force merging, it writes custom metadata as a marker in the cluster state so that we know a force merge is in
+     * process. And it writes another marker in the cluster state upon completion of the force merge.
+     */
+    private void maybeExecuteForceMerge(ClusterState state, DataStream dataStream, Set<Index> indicesToExclude) {
+        List<Index> readOnlyIndices = dataStream.getIndices().stream().filter(index -> indicesToExclude.contains(index) == false).toList();
```

Comment:
read only indices would  imply they have read_only: true. Should we rename this to something else? Maybe just `backingIndices`? 

We could even remove some of the responsibilities of this method and have it just the indices that it attempts to forcemerge?